router_api.go

```diff
@@ -15,8 +31,17 @@ func newAPIHandler(rout *Router) http.Handler {
 			w.WriteHeader(http.StatusMethodNotAllowed)
 			return
 		}
-
-		rout.ReloadRoutes()
+		// Send a message to the reload goroutine, which will start a new timeout
+		// before reloading, or do nothing if one is already in progress.
+		select {
+		case reloadChan <- true:
+			logInfo("router: reload triggered")
+			w.WriteHeader(http.StatusAccepted)
+			w.Write([]byte("Reload triggered"))
+		default:
+			logInfo("router: reload already in progress")
+			w.Write([]byte("Reload already in progress"))
+		}
```

Comment:
Could we make this a channel with a capacity of 1?

On first request, the loop executes immediately and takes about 3 seconds.
In those three seconds, another request is received, a message is sent to the channel and is "queued" for the next loop execution.
Another request is received in those 3 seconds. The channel can't receive this message so the call fails, but there's still a message in the channel.
When the loop finishes, it executes immediately again due to the message in the channel, and clears the channel for future requests.